### ISSUE TITLE
Can't login with the Zoom application if containers are enabled

### ISSUE THREAD

**Issue by jyavenard** [mozilla/multi-account-containers#1847]
This has been an issue for a while.

I use the Work container for any sites related to my work; one of those sites is auth.mozilla.auth0.com which is set to always open in the work container.

When I need to re-sign into Zoom app, I select SSO and it will then re-direct into Firefox into the login page. This will automatically load the authentication page where I can enter my login details.
After that you get redirected to the Zoom.us page.

In the Zoom application you're not signed-in still.

The only way to get the app to be signed-in is to disable the multi-containers extension. Log-in after that you will get prompted to open the Zoom app and you'll get automatically redirected to the zoom-in and you're then authenticated.

For some reasons, the multi-containers break the login process.

**Comment by mijailr** [comment]
This happens when you have to the login to always open on work container and zoom in always open on another container.

The issue here is the callbacks are arriving to the wrong container. Can you try to use the same container with both?

**Comment by jyavenard** [comment]
No, everything is set to use the same work container (zoom.us, sso.mozilla.com and auth0.com)

Authenticating via zoom.us works, authenticating for the zoom external application doesn't.

**Comment by maxxcrawford** [comment]
I'm unable to reproduce this. I have the same workflow: 
- Start in new Work container
- Go to Zoom website (works the same for both `zoom.us` and the `mozilla.zoom.us`
- Click Sign in with SSO
- Add "Mozilla" 
- Triggers Auth, site opens in the same container. 
- Login
- Redirected back to Zoom

All of this happens within the work container. It works as expected. LMK if there's a step I'm missing. 

Note that I have `mozilla.zoom.us` added to always/only open in my Work container.

**Comment by mijailr** [comment]
I have a similar setup but thorough OKTA SSO and I cannot reproduce that behavior, something similar can happens when you have collisions on the default container, for example:

- Github always open on `Personal` container.
- CircleCI always open on `Work` container.
- Try to login on CircleCI using Github SSO.
- Callback opens on work container.
- CircleCI can't get the cookie.

**Comment by jyavenard** [comment]
As described in the first comment, I do not authenticate via the zoom website, but through their app (downloaded from zoom website, I'm on Windows 10).

So you start the zoom application, sign in through that. This will open the SSO page in Firefox. Once authentication has completed instead of being redirected to the zoom external application, you stay in zoom.us site in the browser.

**Comment by dkravetz** [comment]
I am also affected by this issue. I'm running Firefox 79 64 bits on Manjaro, Firefox Multi-Account Containers is version 7.0.2.

I have my work's email (GMail) account in the "Work" container, if I try to login using Google auth via the Zoom Firefox extension, or the Zoom Desktop Client, even when defaulting the `https://accounts.google.com/o/oauth2/v2/auth/oauthchooseaccount` to the "Work" container, the response from Zoom is a 300 error code saying that there was an issue with some parameters.

My workaround for now was to complete this login using the "no container" option.

**Comment by constint** [comment]
Same issue for me with the linux zoom application
I use the same workaround as @dkravetz reopening SSO in "no container"
Quite annoying since I have to remove the "always open in container" options to be allowed to reopen tab in "no container"

**Comment by Nighttraveler** [comment]
I have the same issue with zoom and sometimes with links that require to be logged in (open in new tab, they attempt to open in no container and then try again in work container, seems to be like something is lost in the middle and in work it requires me to login again although im logged in in the Work container). My workaround for the zoom issue in Ubuntu 20.04 was to:

- Change default browser to chromium
- Login in my Gmail/ One Login account
-  Click 'login with google' in zoom app
This will open chromium or the browser of your choice and login in the zoom app.
Then I changed the default app to firefox again.

**Comment by jnbooth** [comment]
I get this issue with all desktop applications that use a web browser to authenticate, where the endpoint they access is inside a container. Discord, GitKraken, etc. Like @Nighttraveler, I have resorted to setting my default browser to chromium, but that's very cumbersome and I don't want to have chromium installed at all.

**Comment by levlozhkin** [comment]
> Note that I have `mozilla.zoom.us` added to always/only open in my Work container.

This was the key for me (and makes this issue a special case of https://github.com/mozilla/multi-account-containers/issues/473). In my case, I needed to add `google.zoom.us` (not just `zoom.us`) to always open in my work container. The SSO redirect chain from Zoom app with Google OAuth2 is roughly:
- google.zoom.us/client_google_signin
- accounts.google.com/o/oauth2
- accounts.google.com/signin/oauth
- google.zoom.us/google/oauth

**Comment by JMoulson** [comment]
The note from @levlozhkin solved this issue for me. Manually adding `google.zoom.us` to the container in question prompted me to open the link in app as normal and complete the app>browser>app authentication flow.
